mypy.ini

```diff
@@ -0,0 +1,12 @@
+[mypy]
+disallow_untyped_defs = True
+disallow_incomplete_defs = True
+check_untyped_defs = True
+warn_no_return = True
+strict_optional = True
+no_implicit_optional = True
+disallow_any_generics = True
+warn_redundant_casts = True
+warn_unused_configs = True
+show_traceback = True
+ignore_missing_imports = True
```

Comment:
Are these a set of recommended defaults? Should we do further work to make the type checking more strict?

Reply:
There is no set of recommended defaults per se. This was originally based on the options mypy uses to check itself (which in turn is based on the options `mypy --strict` enables). I removed a couple of things (such as `disallow_subclassing_any`) which we would trigger, and I just added `disallow_untyped_calls`, after going through the options listed in the docs.

Without providing types for our 3rd party dependencies, I don't see any other options we could turn on that would make checking stricter.